zerver/lib/topic_link_util.py

```diff
@@ -42,29 +44,27 @@ def encode_hash_component(s: str) -> str:
     return "".join(hash_replacements.get(c, c) for c in encoded)
 
 
-def channel_topic_url(stream_id: int, stream_name: str, topic_name: str | None = None) -> str:
-    link = f"#narrow/channel/{stream_id}-{encode_hash_component(stream_name.replace(' ', '-'))}"
-    if topic_name:
-        link += f"/topic/{encode_hash_component(topic_name)}"
-    return link
-
-
 def get_fallback_markdown_link(
     stream_id: int, stream_name: str, topic_name: str | None = None
 ) -> str:
     """
     Generates the markdown link syntax for a stream or topic link.
     """
     escape = escape_invalid_stream_topic_characters
-    url = channel_topic_url(stream_id, stream_name, topic_name)
-    if topic_name:
-        return f"[#{escape(stream_name)} > {escape(topic_name)}]({url})"
+    link = f"#narrow/channel/{stream_id}-{encode_hash_component(stream_name.replace(' ', '-'))}"
+    text = f"#{escape(stream_name)}"
+    if topic_name is not None:
+        link += f"/topic/{encode_hash_component(topic_name)}"
+        if topic_name == "":
+            text += " > " + Message.EMPTY_TOPIC_FALLBACK_NAME
```

Comment:
Should `escape` be being called on `Message.EMPTY_TOPIC_FALLBACK_NAME`? Hmm, or should that be being enclosed in `*`s for italics?